pkg/controller/issuer/certificate/reconciler.go

```diff
@@ -1031,15 +1047,16 @@ func (r *certReconciler) findSecretByHashLabel(namespace string, spec *api.Certi
 				bestNotAfter.Before(cert.NotAfter) ||
 				secretRef != nil && bestNotAfter.Equal(cert.NotAfter) && obj.GetName() == secretRef.Name && core.NormalizeNamespace(obj.GetNamespace()) == core.NormalizeNamespace(secretRef.Namespace) {
 				best = obj
+				notBefore = cert.NotBefore
 				bestNotAfter = cert.NotAfter
 			}
 		}
 	}
 	if best == nil {
-		return nil, "", nil
+		return nil, fmt.Errorf("could not determine a best match in the certificate secrets with the hash: %s", specHash)
```

Comment:
Nothing found is a valid outcome. Should not result in an error for a find method.
```suggestion
		return nil, nil
```

Reply:
Makes sense, thanks! :) 
Fixed via: [`a13a23b` (#489)](https://github.com/gardener/cert-management/pull/489/commits/a13a23b6ddbb97ae3a18092cfeacb3da5d63cc9e)